tests/h/views/admin/features_test.py

```diff
@@ -185,6 +187,26 @@ def test_cohorts_edit_with_users(factories, pyramid_request):
     assert len(result["cohort"].members) == 2
 
 
+def test_cohorts_delete(factories, pyramid_request):
+    cohort = models.FeatureCohort(name="Test Cohort")
+
+    # Add some members to the cohort. This is not essential, but it ensures
+    # that this facility can be used to delete a cohort which still has members.
+    user1 = factories.User(username="benoit")
+    user2 = factories.User(username="emily", authority="foo.org")
+    cohort.members.append(user1)
+    cohort.members.append(user2)
+
+    pyramid_request.db.add(cohort)
+    pyramid_request.db.flush()
+
+    pyramid_request.matchdict["id"] = cohort.id
+    result = cohorts_delete({}, pyramid_request)
+
+    assert cohort in pyramid_request.db.deleted
```

Comment:
Other tests use a variety of methods to check whether a DB entry has been deleted (I searched for `assert.*deleted`). Checking for the object in the SQLAlchemy session's `deleted` set seems like the simplest of those methods.